scripts/glue/load-s3-to-db.py

```diff
@@ -0,0 +1,114 @@
+import os
+import sys
+import boto3
+import logging
+import psycopg2 
+import pandas as pd
+from io import BytesIO, StringIO
+from argparse import ArgumentParser
+from datetime import datetime, timedelta
+
+# Initialize logging
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+
+def load_environment_variables():
+    try:
+        # if this is run in glue, use the AWS Glue provided env variables
+        # for reference: https://docs.aws.amazon.com/glue/latest/dg/aws-glue-api-crawler-pyspark-extensions-get-resolved-options.html
+        from awsglue.utils import getResolvedOptions
+        args = getResolvedOptions(sys.argv, ['DB_HOST', 'DB_PORT', 'POSTGRES_DB', 'POSTGRES_USER', 'POSTGRES_PASSWORD'])
+        return args
+    except Exception:
+        # if run locally, use the .env file 
+        from dotenv import load_dotenv
+        load_dotenv()  # Fallback to .env file if not running on AWS Glue
+        return {
+            'DB_HOST': os.getenv('DB_HOST'),
+            'DB_PORT': os.getenv('DB_PORT'),
+            'POSTGRES_DB': os.getenv('POSTGRES_DB'),
+            'POSTGRES_USER': os.getenv('POSTGRES_USER'),
+            'POSTGRES_PASSWORD': os.getenv('POSTGRES_PASSWORD')
+        }
+
+def is_s3_path(file_path):
+    return file_path.startswith('s3://')
+
+def read_csv(file_path):
+    logging.info("Reading data from %s", file_path)
+    try:
+        if is_s3_path(file_path):
+            s3 = boto3.client('s3')
+            bucket_name = file_path.split('/')[2]
+            s3_path = '/'.join(file_path.split('/')[3:])
+            obj = s3.get_object(Bucket=bucket_name, Key=s3_path)
+            df = pd.read_csv(BytesIO(obj['Body'].read()))
+        else:
+            df = pd.read_csv(file_path)
+    except Exception as e:
+        logging.error("Error reading CSV file: %s", e)
+        sys.exit(1)
+
+    # Remove the 'Unnamed: 0' column if it exists
+    if 'Unnamed: 0' in df.columns:
+        df = df.drop(columns=['Unnamed: 0'])
+
+    return df
+
+def process_data(df):
+    # Convert 'date_posted' from string to datetime
+    df['date_posted'] = pd.to_datetime(df['date_posted'], format='%Y-%m-%d', errors='coerce')
+
+    # Calculate the cutoff date for one month ago
+    one_month_ago = datetime.now() - timedelta(days=30)
+
+    # Filter for the last month's data
+    df = df[df['date_posted'] >= one_month_ago]
+
+    # Sort by 'date_posted'
+    df = df.sort_values(by='date_posted', ascending=False)
+    
+    # Filtering for Jobs in Canada and creating an explicit copy to avoid SettingWithCopyWarning
+    df_canada = df[df['location'].str.contains("Canada", na=False)].copy()
+
+    # Cleaning 'location' Column
+    # Extracting city from the 'location' column
+    df_canada['city'] = df_canada['location'].apply(lambda x: x.split(',')[0].strip())
+
+    return df_canada
+
+def load_to_postgres(df, db_config):
```

Comment:
can you move this file or these queries into a `separate file` in `database` folder?

so we can add unit tests to them later on?